hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -470,7 +470,7 @@ class HoodieSparkSqlWriterInternal {
 
             // Short-circuit if bulk_insert via row is enabled.
             // scalastyle:off
-            if (hoodieConfig.getBoolean(ENABLE_ROW_WRITER) && operation == WriteOperationType.BULK_INSERT) {
+            if (hoodieConfig.getBoolean(ENABLE_ROW_WRITER) && operation == WriteOperationType.BULK_INSERT && !hoodieConfig.getBooleanOrDefault(HoodieWriteConfig.COMBINE_BEFORE_INSERT)) {
```

Comment:
If meta fields are present, then i think we should still allow row writer path for bulk insert. So, how about only disabling row writer when `hoodie.populate.meta.fields` is false and `hoodie.combine.before.insert` is true?

Reply:
Thanks for reply. I will look for this way of resolution.

Reply:
Reverted this change, and resolved issue by disabling row writer if the operation is `BULK_INSERT`, `hoodie.populate.meta.fields` is false, and `hoodie.combine.before.insert`.